src/Mvc/src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/BindingSource.cs

```diff
@@ -196,7 +196,22 @@ public virtual bool CanAcceptDataFrom(BindingSource bindingSource)
                 throw new ArgumentException(message, nameof(bindingSource));
             }
 
-            return this == bindingSource;
+            if (this == bindingSource)
+            {
+                return true;
+            }
+
+            if (this == ModelBinding)
+            {
+                return bindingSource == Form || bindingSource == Path || bindingSource == Query;
+            }
+
+            if (bindingSource == ModelBinding)
```

Comment:
This condition seems really strange. The previous if statement makes intuitive sense, but in combination with it, it suggests that `Path`, `Query` and `Form` are sort of aliases for `ModelBinding`. Could you add a comment that says why this condition is required?

Reply:
Good catch. I'll remove https://github.com/aspnet/AspNetCore/blob/e0de8a31375e173c03ddb5cd7f21bd5bcc252412/src/Mvc/src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/BindingSource.cs#L209-L212